noxfile.py

```diff
@@ -1,28 +1,34 @@
 """
-Configuration file for nox.
+Running `nox` without arguments will run tests with the version of
+Python that `nox` is installed under, skipping slow tests. To invoke a
+nox session, run `nox -s "<session>"`, where <session> is replaced with
+the name of the session. To list available sessions, run `nox -l`.
 
-nox is an automation tool that allows us to configure and perform tasks
-using programmable Python sessions. Each nox session is defined via a
-function decorated with ``@nox.session``. The name of a session is given
-by the name of the function.
+The tests can be run with the following options:
 
-To list available sessions, run `nox -l`. To invoke a nox session, run
-`nox -s <session>`, where <session> is replaced with the name of the
-session.
+* "all": run all tests
+* "skipslow": run tests, except tests decorated with `@pytest.mark.slow`
+* "cov": run all tests with code coverage checks
+* "lowest-direct" : run all tests with lowest version of direct dependencies
 
-Documentation for nox is at: https://nox.thea.codes
+Doctests are run only for the most recent versions of Python and
+PlasmaPy dependencies, and not when code coverage checks are performed.
 """
 
-import nox
+import sys
 
-nox.options.default_venv_backend = "uv|virtualenv"
-nox.options.sessions = []
+import nox
 
-supported_python_versions = ("3.10", "3.11", "3.12")
+supported_python_versions: tuple[str, ...] = ("3.10", "3.11", "3.12")
```

Comment:
I was debating about whether the type hint annotations were necessary in this file, but decided that they were since the annotations will help IDEs like PyCharm to highlight configuration problems while we're still making edits to a file.